Add New Lab Starting Scenario

KyrasRisven opened the pull request:
<!--
### How to use
Leave the headings unless they don't apply to your PR, replace commented out text (surrounded with <!–– and ––>) with text describing your PR.
-->

#### Summary
```SUMMARY: Content "New lab starting scenario added."```

#### Purpose of change
The only way to start in a lab was the lab challenge, which drastically limits your starting professions.

#### Describe the solution
Adds a new scenario starting in a locked lab that allows the choice of any profession, but rewards only 4 points instead of the Lab Challenge's 8. It also removes the ability to start with mutations and the experiment cell starting location.

#### Describe alternatives you've considered
Adding a starting NPC (representing your guide).
Adding classes to the lab challenge, but that would make the 8 points too strong.

#### Additional Context
Flavor wise, I went with the idea of the player having connections that got them into the lab since otherwise it's hard to justify being in a top-secret facility when you aren't a part of it. I have no strong feelings to the justification given and if anyone can think of a reason that is better justified by the lore feel free to change it.

Leland commented on data/json/scenarios/scenarios.json:
```diff
@@ -175,6 +175,16 @@
+    "ident": "lab_normal",
+    "name": "Locked Lab",
+    "points": -4,
+    "description": "Thanks to your connections you were invited to visit a top-secret laboratory. Unfortunately your visit coincided with the cataclysm and now the lab has gone into lockdown. Find a way out or perish.",
```
I believe `cataclysm` should be capitalized. Needs a comma after "unfortunately" as well

```suggestion
    "description": "Thanks to your connections you were invited to visit a top-secret laboratory. Unfortunately, your visit coincided with the Cataclysm and now the lab has gone into lockdown. Find a way out or perish.",
```

ZhilkinSerg replied:
and two whitespaces between sentences

curstwist commented:
I'd be tempted to have this cost points or at least be 0 sum.  Most of the goodies are in the labs, this would ensure an OP start for most normal survivors since the only challenge left is getting out (which has become easier over time).

RarkGrames commented:
I think that this backstory would not be applicable at all for some classes. Professions starting armed to the teeth, professions with animals, farmer with a pitchfork, high school student, guy in a towel or naked, guy in a gimp suit, etc.

My take is that a random portal sucked them in to plop them out in the lab. It is cheesy as hell, but at least it answers why are they standing in the middle of a lab with their stuff from their everyday life or strange situations.

KyrasRisven commented:
The suggestion for the portal is good, it justifies why you wouldn't have had your gear confiscated or anything like that. Might change it to that later. 

As for point cost, scenario costs are already horribly unbalanced so I'm not too concerned on that, reducing it to a two point gain might be more fair. (Also, Helicopter Crash should see the same reduction).

KyrasRisven commented:
Okay so I changed it to the portal backstory as that justifies it far better. I also added the Zombies Nearby flag to add some difficulty and justify the point gain.

FrozenSnowFox commented:
Outside of fun I also don't see all professions working for a lab scenario but here's some ideas that could work as alternate starts if that's your intention.

Lab Security - You work in the lab doing security (security guys were added a while back).
Military - You are at the lab for whatever reason (plenty of military guys in the vaults).
Espionage - Military personal and some specific civilian professions (journalist?) snuck into the lab to find out information on the chaos unfolding in the world.

Amoebka commented:
I think having a lab start that adds more professions and disallows mutations is needed, but it shouldn't give points and shouldn't unlock ALL professions. Some professions that would make sense for a non-mut lab start are:

Chef, Electrician, Augmentation Associate (civilians working in the lab)

Security Guard, Military Recruit, Bionic Soldier (all these have zombiefied equivalents spawning in labs)

Bionic Monster (an easier spin on the broken cyborg)

TechyBen commented:
If portal traps is the go to option for a bit of fun and rnd... can we make it true rnd? As in, portal traps are in the game.

So "You stepped on something that went 'click' and a portal surround you. Where have you been taken?"

But the limit being no choice on where to spawn. Could be in a lab. Or in a town. Or anywhere. Surrounded start plus the spoilers infection to the player. Could even have some items missing or radiation exposure. Getting trapped in and experimental and dangerous teleporter cannot be good for your health.

I-am-Erk commented:
The portals we have currently are to the netherum, a hellish between-space. We haven't established anything about portals leading you around the same dimension, and that opens up a can of worms I don't think we should open

KyrasRisven commented:
So to be clear, the random short range teleport that happens when you enter the same square as a portal isn't meant to be a canon thing and is just a gameplay oversight?

Mecares commented:
Well they work if you step on them and are short range, not a good explanation for anybody getting into the lowers levels of a lab.

KyrasRisven commented:
I'll move this to a mod when I have time. Is the Classes and Scenarios mod meant to take contributions or do I need to make a separate mod?

Also going to start researching mapping because the lab is my favorite area and I want to spice it up.

RarkGrames commented:
@I-am-Erk This portal thing was just one quick thought popping in my mind, I didn't consider this thing very thoroughly. I kind of regretted it, seeing all this discussion here. I agree with you that it should be a mod. Professions starting with guns just blaze through the lab, snipe the bots and turrets, scoop up valuable stuff and the card at the end, and leave with some extra points and loot after an hour. Takes the challenge out of "lab challenge". It is easier actually, because you don't have wandering hordes spawning on you because of the shooting. This is kind of the Infected challenge starting in an empty hospital.
Many things got moved to mods for less radical things, so this should be also a mod.

Fris0uman commented:
The original idea of having connection getting you in for a not so officially approved visit is ok I think if it only applies to a restricted number of profession.

l29ah commented:
I wonder when Blackbelt will have to go...

ZhilkinSerg commented:
Can you resolve conflicts and react to Eric's to review please?

I-am-Erk commented:
Closed due to inactivity, please comment if you're around and would like it reopened.
